### PR TITLE
:seedling: CRD labeling is done now by k8s api to fix gosec issues

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
**What this PR does / why we need it**:
 This PR is doing CRD labeling with k8s golang api instead of running shell command.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
This PR is added to fix for gosec issues that we have in capm3.
Fixes #

### CODE REVIEW

File: test/e2e/pivoting.go [hunk 1]
```diff
@@ -392,50 +393,22 @@ func RemoveDeployment(ctx context.Context, inputGetter func() RemoveDeploymentIn
...
 
-func labelBMOCRDs(targetCluster framework.ClusterProxy) {
-	labels := []string{
-		"clusterctl.cluster.x-k8s.io=",
-		"cluster.x-k8s.io/provider=metal3",
-	}
-	kubectlArgs := ""
-	if targetCluster != nil {
-		kubectlArgs = fmt.Sprintf("--kubeconfig=%s", targetCluster.GetKubeconfigPath())
-	}
```
[the remark this answers is not in the document]
Comment (adilGhaffarDev) [reply test/e2e/pivoting.go:403]: now we are passing the client as a parameter and using that `labelBMOCRDs(ctx, input.TargetCluster.GetClient())`
I am testing if its working as expected on not

**Review by lentzi90 — COMMENTED** [review]
Looks good! Just a couple of nits below.

File: test/e2e/common.go [hunk 2]
```diff
@@ -745,3 +746,24 @@ func (Metal3LogCollector) CollectMachinePoolLog(_ context.Context, _ client.Clie
 func (Metal3LogCollector) CollectInfrastructureLogs(_ context.Context, _ client.Client, _ *clusterv1.Cluster, _ string) error {
 	return fmt.Errorf("CollectInfrastructureLogs not implemented")
 }
+
+func LabelCRD(ctx context.Context, cs *clientset.Clientset, crdName string, labels map[string]string) error {
```
[the remark this answers is not in the document]
Comment (adilGhaffarDev) [reply test/e2e/common.go]: fixed the suggestions kindly check.

File: test/e2e/pivoting.go [hunk 3]
```diff
@@ -392,50 +395,30 @@ func RemoveDeployment(ctx context.Context, inputGetter func() RemoveDeploymentIn
...
-		var cmd *exec.Cmd
-		if kubectlArgs == "" {
-			cmd = exec.Command("kubectl", "label", "--overwrite", "crds", crdName, label) //#nosec G204:gosec
-		} else {
-			cmd = exec.Command("kubectl", kubectlArgs, "label", "--overwrite", "crds", crdName, label) //#nosec G204:gosec
-		}
-		err := cmd.Run()
-		Expect(err).To(BeNil(), "Cannot label BMO CRDs")
-	}
+	config, err := clientcmd.BuildConfigFromFlags("", targetCluster.GetKubeconfigPath())
```
Comment (mboukhalfa) [inline test/e2e/pivoting.go:403]:
```suggestion
	targetClusterClient := targetCluster.GetClient()
```
Might be easier if that does the same job

**Review by lentzi90 — COMMENTED** [review]
/approve

File: test/e2e/pivoting.go (hunk 3 above)
Comment (adilGhaffarDev) [reply test/e2e/pivoting.go]: I have changed it kindly check.

**Pull request merged**